launch-scripts/launch_gapbs.py

```diff
@@ -102,41 +115,73 @@
     num_cpus = ['1', '2', '4']
     workloads = ['bc', 'bfs', 'cc', 'sssp', 'tc','pr']
     sizes = ['10', '15', '20']
+    cpu_types = ['kvm', 'atomic', 'simple', 'o3']
     graphs = ['roadU.sg', 'webU.sg']
-    mem_types = ['classic', 'MI_example']
+    mem_types = ['classic', 'MI_example','MESI_Two_Level']
 
-    for num_cpu in num_cpus:
-        for workload in workloads:
-            for size in sizes: 
-                for mem in mem_types:
-                    run = gem5Run.createFSRun(
-                        'Running GAPBS',
-                        'gem5/build/X86/gem5.opt',
-                        'configs-gapbs-tests/gapbs_config.py',
-                        'results/run_exit/vmlinux-5.2.3/gapbs/kvm/{}/{}/{}/synthetic/{}'.
-                        format( num_cpu, mem ,workload, size),
-                        gem5_binary, gem5_repo, experiments_repo,
-                        'linux-stable/vmlinux-5.2.3',
-                        'disk-image/gapbs-image/gapbs',
-                        linux_binaries, disk_image, 'kvm', num_cpu, mem ,workload, '1', size,
-                        timeout = 6*60*60 
-                        )
-                    run_gem5_instance(run,)
-                            
-    for num_cpu in num_cpus:
-        for workload in workloads:
-            for graph in graphs: 
-                for mem in mem_types:
-                    run = gem5Run.createFSRun(
-                        'Running GAPBS',
-                        'gem5/build/X86/gem5.opt',
-                        'configs-gapbs-tests/gapbs_config.py',
-                        'results/run_exit/vmlinux-5.2.3/gapbs/kvm/{}/{}/{}/realgraph/{}'.
-                        format( num_cpu, mem ,workload, graph),
-                        gem5_binary, gem5_repo, experiments_repo,
-                        'linux-stable/vmlinux-5.2.3',
-                        'disk-image/gapbs-image/gapbs',
-                        linux_binaries, disk_image, 'kvm', num_cpu, mem ,workload, '0', graph,
-                        timeout = 6*60*60 
-                        )
-                    run_gem5_instance(run,)
+    for cpu in cpu_types:
```

Comment:
See https://github.com/darchr/gem5art-experiments/pull/39/files#diff-5a16318b0c26682857baf1ff61bae5e4R119 for a much cleaner way to write this code.

Basically, you can replace all of the nested for loops with a single line `starmap(<func>, num_cpus, workloads, sizes, mem_types)` which will create all of the runs for you (assuming that `<func>` returns a run). Ask Ayaz if you have questions here. And feel free to ask me, too :). See https://docs.python.org/3/library/itertools.html#itertools.starmap